src/anniversary/sf33rd/Source/Game/bg.c

```diff
@@ -61,17 +260,153 @@ INCLUDE_RODATA("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", literal_502
+                rw_dat[i].rw_cnt = *rw_dat[i].rwd_ptr++;
+                rw_dat[i].gbix = *rw_dat[i].rwd_ptr++;
+            } else {
+                rw_dat[i].rw_cnt = *rw_dat[i].rwd_ptr++;
+                rw_dat[i].gbix = *rw_dat[i].rwd_ptr++;
+            }
+        }
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", bgDrawOneScreen);
+        i++;
+    }
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", bgDrawOneChip);
+void bgDrawOneScreen(s32 bgnum, s32 gixbase, s32 *xx, s32 *yy, s32 /* unused */, s32 ofsPal, PPGDataList *curDataList) {
+    s32 i, x, y, gbix;
+
+    y = yy[0];
+    while (y < yy[1]) {
+        x = xx[0];
+        while (x < xx[1]) {
+            gbix = ((y >> 7) << 3) + (x >> 7) + gixbase;
+
+            if (rw_bg_flag[bgnum] && rw_num) {
+                for (i = 0; i < rw_num; i++) {
+                    if (bgnum == rw_dat[i].bg_num && gbix == rw_dat[i].rwgbix) {
+                        gbix = rw_dat[i].gbix;
+                        if (!(ppgCheckTextureNumber(0, gbix))) {
+                            ppgSetupCurrentDataList(&ppgRwBgList);
+                        }
+                        break;
+                    }
+                }
+            }
+
+            bgDrawOneChip(x, y, 128, 128, gbix, -1, ofsPal);
+            ppgSetupCurrentDataList(curDataList);
+            x += 128;
+        }
+        y += 128;
+    }
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", bgAkebonoDraw);
+void bgDrawOneChip(s32 x, s32 y, s32 xs, s32 ys, s32 gbix, u32 vtxCol, s32 ofsPal) {
+    s32 i;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", ppgCalScrPosition);
+    if (No_Trans == 0 && ppgCheckTextureNumber(0, gbix)) {
+        ppgCalScrPosition(x, y, xs, ys);
+        if (!(scrDrawPos->x < 384.0f && !(scrDrawPos[3].x < 0.0f) && (scrDrawPos->y < 224.0f) &&
+              !(scrDrawPos[3].y < 0.0f))) {
+            return;
+        } else {
+            ppgWriteQuadUseTrans(scrDrawPos, vtxCol, 0, gbix, 0, 0, ofsPal);
+        }
+    }
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", scr_trans_sub2);
+void bgAkebonoDraw() {
+    s32 i;
+
+    scrDrawPos->x = 0.0f;
+    scrDrawPos->y = 0.0f;
+    scrDrawPos[3].x = 128.0f;
+    scrDrawPos[3].y = 224.0f;
+    scrDrawPos->z = scrDrawPos[3].z = *(&PrioBase[bg_priority[3]]);
+    scrDrawPos->s = scrDrawPos->t = 0.0f;
+    scrDrawPos[3].s = 1.0f;
+    scrDrawPos[3].t = 0.875f;
+    i = 0;
+
+    while (i < 3) {
+        ppgWriteQuadUseTrans(scrDrawPos, -1U, NULL, i, i, 0, 0);
+        scrDrawPos->x += 128.0f;
+        scrDrawPos[3].x += 128.0f;
+        i++;
+    }
+}
+
+void ppgCalScrPosition(s32 x, s32 y, s32 xs, s32 ys) {
+    Vec3 point[2];
+
+    point[0].x = (f32)x;
+    point[0].y = (f32)y;
+    point[1].x = (f32)(x + xs);
+    point[1].y = (f32)(y + ys);
+    point[0].z = point[1].z = 0;
+    njCalcPoints(0, &point[0], &point[0], 2);
```

Comment:
Args 2 and 3 have array semantics, so it makes sense to treat parameters as arrays too
```suggestion
    njCalcPoints(0, point, point, 2);
```